pkg/ingester/ingester.go

```diff
@@ -2405,20 +2406,25 @@ func (i *Ingester) shipBlocks(ctx context.Context, allowed *util.AllowedTenants)
 }
 
 func (i *Ingester) compactionLoop(ctx context.Context) error {
-	// Apply a jitter on the first head compaction
-	firstHeadCompaction := true
-	ticker := time.NewTicker(util.DurationWithPositiveJitter(i.cfg.BlocksStorageConfig.TSDB.HeadCompactionInterval, initialHeadCompactionJitter))
+	infoFunc := func() (int, int) {
+		if i.cfg.LifecyclerConfig.RingConfig.ZoneAwarenessEnabled {
+			zones := i.lifecycler.Zones()
+			if len(zones) != 0 {
+				return slices.Index(zones, i.lifecycler.Zone), len(zones)
+			}
+		}
+
+		// Lets create the slot based on the hash id
+		i := int(client.HashAdd32(client.HashNew32(), i.lifecycler.ID) % 10)
+		return i, 10
```

Comment:
Is there a usecase where we want to change the number of slots 10?

Reply:
humm...
idk.. maybe if the head compaction is very large we may benefit from more than 10 slots... But we are limiting the head compaction to 30 min so those are 3 min slots... and will be jittered inside as well.. so i think its ok